OpenSim/Common/Component.h

```diff
@@ -327,6 +324,8 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
      * must be done before calling this method on the top model. */
     template <typename T = Component>
     ComponentList<T> getComponentList() const {
+        Component* mutableThis = const_cast<Component*>(this);
+        mutableThis->initComponentTreeTraversal(*mutableThis);
```

Comment:
Can this be cached? The method here could be `mutableThis->calcComponentListIfNeeded()` or whatever. Maybe you could write it that way now and add the actual caching and invalidating later?